pkg/kube/client.go

```diff
@@ -809,7 +809,7 @@ func (c *Client) watchUntilReady(timeout time.Duration, info *resource.Info) err
 
 	ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), timeout)
 	defer cancel()
-	_, err = watchtools.UntilWithoutRetry(ctx, w, func(e watch.Event) (bool, error) {
+	_, err = watchtools.ListWatchUntil(ctx, w, func(e watch.Event) (bool, error) {
```

Comment:
Based on the test failure output, it looks like the watcher we create with the resource helper isn't the right type to use with this function. You'll probably need to do something to convert the watch interface to the `cache.ListerWatcher` it expects

Reply:
@thomastaylor312
After a bit of fiddling I think I've got this right. CI run is now green.